example/loop_advection/loop_advection_package.cpp

```diff
@@ -0,0 +1,368 @@
+
+  using A_face = E_edge;
+
+  //Initialize vector potential at cell edges
+  const auto &A = md->PackVars(std::vector<std::string>{"E_edge"});
+  for( int edge_dir=1; edge_dir <= 3; edge_dir++){
+    CartDir edge_dir_enum = static_cast<CartDir>(edge_dir);
+
+    IndexRange ib = pmb->cellbounds.GetEdgeBoundsI(IndexDomain::entire,edge_dir_enum);
+    IndexRange jb = pmb->cellbounds.GetEdgeBoundsJ(IndexDomain::entire,edge_dir_enum);
+    IndexRange kb = pmb->cellbounds.GetEdgeBoundsK(IndexDomain::entire,edge_dir_enum);
+
+    pmb->par_for(
+      "LoopAdvection::ProblemGenerator::init_A.x"+(edge_dir)+"e", 0, vf.dim(5)-1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        const auto &coords = v.GetCoords(b);
+        const Real r = sqrt( x*x  + y*y + z*z );
+
+
+        const Real x = (edge_dir == 1) ? coords.x1v(i) : coords.x1f(i);
+        const Real y = (edge_dir == 2) ? coords.x2v(j) : coords.x2f(j);
+        const Real z = (edge_dir == 3) ? coords.x3v(k) : coords.x2f(k);
+
+        const Real ax = 0;
+        const Real ay = 0;
+        const Real az = max( amp*(R - r),0);
+
+        v( edge_dir, b, A_face(), k, j, i) = (edge_dir == 1 ) ? ax :
+                                             (edge_dir == 2 ) ? ay : az;
+      });
+  }
+
+
+  //Initialize magnetic field from curl of vector potential
+  //Computing each direction individually (not necessarily the fastest method)
+  for( int face_dir=1; face_dir <= 3; face_dir++){
+    CartDir face_dir_enum = static_cast<CartDir>(face_dir);
+
+    IndexRange ib = pmb->cellbounds.GetFaceBoundsI(IndexDomain::entire,face_dir_enum);
+    IndexRange jb = pmb->cellbounds.GetFaceBoundsJ(IndexDomain::entire,face_dir_enum);
+    IndexRange kb = pmb->cellbounds.GetFaceBoundsK(IndexDomain::entire,face_dir_enum);
+
+    pmb->par_for(
+      "LoopAdvection::ProblemGenerator::init_B.x"+(face_dir)+"f", 0, vf.dim(5)-1, kb.s, kb.e, jb.s, jb.e, ib.s, ib.e,
+      KOKKOS_LAMBDA(const int k, const int j, const int i) {
+        const auto &coords = v.GetCoords(b);
+
+        //Compute the Curl of E
+        const int cart_dir_p1 = (cart_dir    )%3 + 1;
+        const int cart_dir_p2 = (cart_dir + 1)%3 + 1;
+
+        const Real dp1 = (cart_dir == 1) ? coords.dx1(i) :
+                        +(cart_dir == 2) ? coords.dx2(j) :
+                                           coords.dx3(k);
+        const Real dp2 = (cart_dir == 1) ? coords.dx1(i) :
+                        +(cart_dir == 2) ? coords.dx2(j) :
+                                           coords.dx3(k);
+
+        //First term: dEzdy, dExdz, or dEydx
+        const Real dEp2dp1 =(v( cart_dir_p2, b, A_face(),  k, j, i)
+                            -v( cart_dir_p2, b, A_face(),  k - (cart_dir_p1 == 3),
+                                                           j - (cart_dir_p1 == 2),
+                                                           i - (cart_dir_p1 == 1)))/dp1;
```

Comment:
I mentioned this in #692 - I also find the ordering with mesh block awkward as well, but I think it should go after the variable idx since the mesh block refers to a location in space much like `k`, `j`, `i`. If I could rewrite things, I would use `v( A_face(cart_dir_p2), b, k, j, i)` to group the variable indices and spatial indices together.

I like the idea of using the template name machinery, although I don't understand it well enough to implement it like that.

Reply:
I strongly prefer the meshblock index `b` to be the outermost index, as it should be slowest-moving, even slower than the face location IMO. I think we're in agreement consistency would be best, though. ;)

I bet it wouldn't be too hard to generalize the face name stuff, right @lroberts36 ? But we should probably wait on that until the data structures for faces/nodes/edges have stabilized.

Reply:
Remembering to when I wrote this, there is no way to be consistent with the previous flux implementation since there is no direct access to fluxes from a `MeshBlockPack`.  For a `MeshBlockPack`s you have to pull out a `VariableFluxPack` and access the fluxes from it. So I don't think there are any consistency issues (contrary to anything I may or may not have said previously). I tend to agree with @Yurlungur that the block index probably is the most logical thing to have out front.